metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/detection/ConnectorStagingManager.java

```diff
@@ -0,0 +1,338 @@
+package org.sentrysoftware.metricshub.engine.strategy.detection;
+
+/*-
+ * ╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲
+ * MetricsHub Engine
+ * ჻჻჻჻჻჻
+ * Copyright 2023 - 2024 Sentry Software
+ * ჻჻჻჻჻჻
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱
+ */
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.sentrysoftware.metricshub.engine.connector.model.Connector;
+import org.sentrysoftware.metricshub.engine.connector.model.ConnectorStore;
+
+/**
+ * This class stages connector identifiers based on configured directives.
+ * It facilitates the identification of connectors for automatic detection and forced inclusion or exclusion.
+ */
+@Slf4j
+public class ConnectorStagingManager {
+
+	/**
+	 * Regular expression pattern for parsing configured connector directives.
+	 */
+	private static final Pattern CONFIGURED_CONNECTOR_PATTERN = Pattern.compile("^((?!-#)([#\\+-]?)|((-#)?))(.+)$");
+
+	/**
+	 * Regular expression pattern for matching included connectors for automatic detection.
+	 */
+	private static final Pattern INCLUDED_CONNECTORS_PATTERN = Pattern.compile("^(?![+-])(#?)(.+)$");
+
+	/**
+	 * The host name of the resource we currently monitor
+	 */
+	private String hostname;
+
+	/**
+	 * Whether logging is enabled for this manager.
+	 */
+	private boolean enableLogging;
```

Comment:
It would be better to name it `hasLoggingEnabled`

Reply:
I could name this member isLoggingEnabled